### ISSUE TITLE
interpolate in torch convert into onnx, then into ncnn. Unsupport upsample scale.

### ISSUE THREAD

**Issue by ThomsonW** [Tencent/ncnn#2407]
hi there, thanks for your great job.
i am working on converting a slim centernet-like model trained by pytorch into ncnn model. According to provided doc, i first convert .pth model into .onnx by torch whose opset is defalut version 9. but i can not use convertor tool you provided in /tools/onnx/onnx2ncnn to convert into >bin and .params.
the operator interpolate is used in my model to upsample with factor 2. when i execute the convert tool, the error occurred like that:
```
Cast not supported yet!
  # to=1
Unknown data type 0
Unsupported Upsample scales !

```  
i checked related issue but no solution yet.

what's more, how does the cast operator problem?  there is no cast operator in my model.

PS. i looked through the code of the tool, line 3847:
```

            float h_scale = 1.f;
            float w_scale = 1.f;
            if (scales.size() == 2)
            {
                w_scale = scales[1];
            }
            else if (scales.size() == 3)
            {
                h_scale = scales[1];
                w_scale = scales[2];
            }
            else if (scales.size() == 4)
            {
                h_scale = scales[2];
                w_scale = scales[3];

                if (scales[1] != 1.f)
                    fprintf(stderr, "Unsupported Upsample scales !\n");
            }
            else
            {
                fprintf(stderr, "Unsupported Upsample scales !\n");
            }
```
the factor 2 is supported according to the code.
how should i do to solve such problem? should i use deconv to replace the interpolate operator? and is there a list of supported operation both in ncnn and onnx/pytorch?
thank you very much.

**Comment by ThomsonW** [comment]
i have solved this problem.
the correct expression is:
```
import torch.nn.functional as F
out = F.interpolate(output2, size=[240,240], mode="nearest")
```

the wrong expression is:
```
import torch.nn.functional as F
out = F.interpolate(output2, size=[output1.size(2), output1.size(3)], mode="nearest")
```

the size should be set directly.

**Comment by cape-zck** [comment]
> 我已经解决了这个问题。 正确的表达是：
> 
> ```
> import torch.nn.functional as F
> out = F.interpolate(output2, size=[240,240], mode="nearest")
> ```
> 
> 错误的表达是：
> 
> ```
> import torch.nn.functional as F
> out = F.interpolate(output2, size=[output1.size(2), output1.size(3)], mode="nearest")
> ```
> 
> 大小应该直接设置。



> i have solved this problem. the correct expression is:
> 
> ```
> import torch.nn.functional as F
> out = F.interpolate(output2, size=[240,240], mode="nearest")
> ```
> 
> the wrong expression is:
> 
> ```
> import torch.nn.functional as F
> out = F.interpolate(output2, size=[output1.size(2), output1.size(3)], mode="nearest")
> ```
> 
> the size should be set directly.

Hello,excuse me, have you completed the deployment of CenterNet on NCNN? I have been researching this for many days, but the results of the deployment have a large loss. Is it convenient to communicate?

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
